### PR TITLE
fix: dependence in  package.xml

### PR DESCRIPTION
**Pull request by h-suzuki-isp**
## Description
Fixed the dependency to enable pybind11 to locate Python modules.

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by ymski — DISMISSED** [review]
LGTM

File: CARET_analyze_cpp_impl/CMakeLists.txt [hunk 1]
```diff
@@ -9,9 +9,10 @@ find_package(ament_cmake REQUIRED)
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
 
 find_package(pybind11_vendor REQUIRED)
-find_package(pybind11 REQUIRED)
+find_package(pybind11 CONFIG REQUIRED)
 find_package(yaml_cpp_vendor REQUIRED)
 
+
```
Comment (ymski) [inline CARET_analyze_cpp_impl/CMakeLists.txt:15]:
```suggestion

```
Please remove this line.

File: CARET_analyze_cpp_impl/CMakeLists.txt [hunk 2]
```diff
@@ -9,9 +9,10 @@ find_package(ament_cmake REQUIRED)
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
 
 find_package(pybind11_vendor REQUIRED)
-find_package(pybind11 REQUIRED)
+find_package(pybind11 CONFIG REQUIRED)
```
Comment (ymski) [inline CARET_analyze_cpp_impl/CMakeLists.txt:12]: Why did you add the `CONFIG` option?

Comment (h-suzuki-isp) [reply CARET_analyze_cpp_impl/CMakeLists.txt]: Fixed in https://github.com/tier4/caret_analyze_cpp_impl/pull/198/commits/d2a1afcf18f98b29701904c9e240b24a1f586f69

**Review by ymski — APPROVED** [review]
LGTM
